### DISCUSSION TITLE
running_mean should contain 68 elements not 75

### DISCUSSION THREAD

**Discussion by YuktiADY** [open-mmlab/mmaction2#2357] [Q&A]
Hello Team,

Hope you are doing well.

I am using STGCN for 2D and 3D action recognition.

For 2D action recognition, I was training on my custom dataset. So I converted my dataset to the pkl file format. It ran successfully.

Now I am doing 3D action recognition. So I converted my custom dataset to pkl file format according to [ntu60_val](https://download.openmmlab.com/mmaction/posec3d/ntu60_xsub_val.pkl).  For 3D action too the pickle file has to be in the same format as 2D? In case pickle file is different for 3D action recognition, where i can find the pickle file for 3D action because i tried to find but was unable to find the file.??

The config i used is this /mmaction2/configs/skeleton/stgcn/stgcn_80e_ntu60_xsub_keypoint_3d.py

This is how my pickle file looks  acc to ntu60_val like after conversion which i pass in the config 

{'frame_dir': 'xxx',
  'label': 2,
  'img_shape': (1200, 1200),
  'original_shape': (1200, 1200),
  'total_frames': 60,
  'keypoint': array([[[[-0.5752752 ,  1.5449048 , -0.35333407],
           [-0.45848957,  2.122168  , -0.7783565 ],
           [-0.56915903,  2.5114708 , -0.58006483],
           ...,
           [-1.0122216 , -0.09795046,  0.48847854],
           [-0.14722271,  1.5622706 , -0.28381488],
           [-0.14523879,  0.97209   , -0.12902533]],
  
          [[-0.9680816 ,  1.5107924 ,  0.16610162],
           [-0.4433155 ,  2.05 [-1.0122216 , -0.09795046,  0.48847854],
           [-0.14722271,  1.5622706 , -0.28381488],
           [-0.14523879,  0.97209   , -0.12902533]],
  
          [[-0.9680816 ,  1.5107924 ,  0.16610162],
           [-0.4433155 ,  2.0596356 , -0.62397057],
           [-0.677656  ,  2.3144732 , -0.35305595],
           ...,
           [-1.2954284 ,  0.23378837,  0.3796356 , -0.62397057],
           [-0.677656  ,  2.3144732 , -0.35305595],
           ...,
           [-1.2954284 ,  0.23378837,  0.37575948],
           [-0.2388921 ,  1.5274153 , -0.64014703],
           [-0.3291791 ,  0.66661656, -0.23001157]],
  
          [[-0.61705935,  1.555523  , -0.08190133],
           [-0.52065563,  2.1946764 , -0.34297913],
           [-0.5824748 ,  2.3736606 , -0.96034425],
           ...,
           [-0.93899643,  0.2939278 ,  0.3628401 ],
           [-0.19814743,  1.8153175 , -0.2928841 ],
           [-0.09000301,  0.8357382 , -0.19873522]],
  
          ...,
  
          [[-0.14083281,  1.5224657 , -0.4185819 ],
           [-0.6803411 ,  1.8933822 , -0.557123  ],
           [-0.606689  ,  2.0906188 , -0.49329972],
           ...,
           [-0.6312376 ,  0.25083363, -0.2258133 ],
           [ 0.12164785,  1.3438213 , -0.46219108],
           [-0.21164745,  0.6044779 , -0.26284647]],
  
          [[-0.06871447,  1.6594003 , -0.28470206],
           [-0.60743654,  1.9643098 , -0.7354354 ],
           [-0.3361878 ,  2.3228328 , -0.54341376],
           ...,
           [-0.8241794 ,  0.09567773, -0.13895753],
           [ 0.02210824,  1.4060467 , -0.50307167],
           [-0.02286317,  0.9342915 , -0.30795985]],
  
          [[-0.17271361,  1.6512171 , -0.6391277 ],
           [-0.7216524 ,  2.121202  , -0.6021455 ],
           [-0.43033436,  2.3559988 , -0.5735959 ],
           ...,
           [-0.9997901 , -0.14000493,  0.11607018],
           [-0.04694979,  1.5226002 , -0.34700772],
           [-0.24039742,  0.8633846 , -0.08101283]]]], dtype=float32),
  'keypoint_score': array([[[1., 1., 1., ..., 1., 1., 1.],
          [1., 1., 1., ..., 1., 1., 1.],
          [1., 1., 1., ..., 1., 1., 1.],
          ...,
          [1., 1., 1., ..., 1., 1., 1.],
          [1., 1., 1., ..., 1., 1., 1.],
          [1., 1., 1., ..., 1., 1., 1.]]], dtype=float16)},

Traceback :
Traceback (most recent call last):
  File "/home/yukti/mmaction2/tools/train.py", line 222, in <module>
    main()
  File "/home/yukti/mmaction2/tools/train.py", line 210, in main
    train_model(
  File "/home/yukti/mmaction2/mmaction/apis/train.py", line 232, in train_model
    runner.run(data_loaders, cfg.workflow, cfg.total_epochs, **runner_kwargs)
  File "/home/yukti/miniconda3/envs/demo/lib/python3.10/site-packages/mmcv/runner/epoch_based_runner.py", line 136, in run
    epoch_runner(data_loaders[i], **kwargs)
  File "/home/yukti/miniconda3/envs/demo/lib/python3.10/site-packages/mmcv/runner/epoch_based_runner.py", line 53, in train
    self.run_iter(data_batch, train_mode=True, **kwargs)
  File "/home/yukti/miniconda3/envs/demo/lib/python3.10/site-packages/mmcv/runner/epoch_based_runner.py", line 31, in run_iter
    outputs = self.model.train_step(data_batch, self.optimizer,
  File "/home/yukti/miniconda3/envs/demo/lib/python3.10/site-packages/mmcv/parallel/data_parallel.py", line 77, in train_step
    return self.module.train_step(*inputs[0], **kwargs[0])
  File "/home/yukti/mmaction2/mmaction/models/skeleton_gcn/base.py", line 155, in train_step
    losses = self(skeletons, label, return_loss=True)
  File "/home/yukti/miniconda3/envs/demo/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/yukti/mmaction2/mmaction/models/skeleton_gcn/base.py", line 106, in forward
    return self.forward_train(keypoint, label, **kwargs)
  File "/home/yukti/mmaction2/mmaction/models/skeleton_gcn/skeletongcn.py", line 16, in forward_train
    x = self.extract_feat(skeletons)
  File "/home/yukti/mmaction2/mmaction/models/skeleton_gcn/base.py", line 121, in extract_feat
    x = self.backbone(skeletons)
  File "/home/yukti/miniconda3/envs/demo/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/yukti/mmaction2/mmaction/models/backbones/stgcn.py", line 274, in forward
    x = self.data_bn(x)
  File "/home/yukti/miniconda3/envs/demo/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1130, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/yukti/miniconda3/envs/demo/lib/python3.10/site-packages/torch/nn/modules/batchnorm.py", line 168, in forward
    return F.batch_norm(
  File "/home/yukti/miniconda3/envs/demo/lib/python3.10/site-packages/torch/nn/functional.py", line 2438, in batch_norm
    return torch.batch_norm(
**RuntimeError: running_mean should contain 68 elements not 75**

Can you please help me how can i resolve this bug? I am doing this work for my  Master's Thesis work. I am stuck at this problem, so that i can proceed further as there is a deadline.

Awaiting for your response.

Thanks & Regards,
Yukti dya

**Comment by cir7** [comment]
[This](https://discuss.pytorch.org/t/runtimeerror-running-mean-should-contain-64-elements-not-96/30846) is the first result given by Google search, maybe it can solve your problem.

**Reply by YuktiADY** [reply]
@cir7 
I will look into the above link and check. Also,
Could you please confirm the pkl file for StGCN 3d action recognition. Since in the config it uses rgb d_60.
Does it uses same pkl file as 2D or its different one because for my custom dataset I have changed in that format in order to paas in config ??
So just wanted to be sure this error is not because of  pkl file ?

**Comment by cir7** [comment]
Please note that anno files in [stgcn_3d](https://github.com/open-mmlab/mmaction2/blob/0.x/configs/skeleton/stgcn/stgcn_80e_ntu60_xsub_keypoint_3d.py#L17) and [stgcn_2d](https://github.com/open-mmlab/mmaction2/blob/0.x/configs/skeleton/stgcn/stgcn_80e_ntu60_xsub_keypoint.py#L17) are different.   
Currently, we don't provide 3D anno for 0.x branch (previous master), you can try main branch, which has available [3D anno files](https://github.com/open-mmlab/mmaction2/blob/main/tools/data/skeleton/README.md#prepare-annotations).

**Reply by YuktiADY** [reply]
Hello @cir7 

I have just few questions .

The main branch provided both 2D and  3D annotations so i can use main to do both 2D and 3D action recogntion ?

I have cloned or downloaded the repository . when i check using  git branch it shows 'master'

Since i have to do 3d action and 2d action recogntion. How can i upgrade to main branch .

Can i do in the same folder where i cloned the repo with this command.
git clone --branch main https://github.com/open-mmlab/mmaction2.git

After doing when i check git branch it shows branch 'main'

When i run my training script
python tools/train.py configs/skeleton/stgcn/stgcn_8xb16-joint-u100-80e_ntu60-xsub-keypoint-3d.py --work-dir ./work_dirs/stgcn_80e_ntu60_xsub_keypoint_3d/test --seed 0 --deterministic --cfg-options  gpu_ids=[0]

I get this error:

Traceback
raceback (most recent call last):
File "/home/yukti/mmaction2/mmaction2/tools/train.py", line 135, in
main()
File "/home/yukti/mmaction2/mmaction2/tools/train.py", line 128, in main
runner = Runner.from_cfg(cfg)
File "/home/yukti/miniconda3/envs/demo_copy1/lib/python3.10/site-packages/mmengine/runner/runner.py", line 431, in from_cfg
runner = cls(
File "/home/yukti/miniconda3/envs/demo_copy1/lib/python3.10/site-packages/mmengine/runner/runner.py", line 385, in init
self.visualizer = self.build_visualizer(visualizer)
File "/home/yukti/miniconda3/envs/demo_copy1/lib/python3.10/site-packages/mmengine/runner/runner.py", line 767, in build_visualizer
return VISUALIZERS.build(visualizer)
File "/home/yukti/miniconda3/envs/demo_copy1/lib/python3.10/site-packages/mmengine/registry/registry.py", line 548, in build
return self.build_func(cfg, *args, **kwargs, registry=self)
File "/home/yukti/miniconda3/envs/demo_copy1/lib/python3.10/site-packages/mmengine/registry/build_functions.py", line 100, in build_from_cfg
raise KeyError(
**KeyError: 'ActionVisualizer is not in the visualizer registry. Please check whether the value of ActionVisualizer is correct r it was registered as expected. More details can be found at https://mmengine.readthedocs.io/en/latest/advanced_tutorials/config.html#import-the-custom-module'**



I checked the mmengine version its 0.6.0

Awaiting your response.

**Comment by cir7** [comment]
1. Main branch supports both 2D and 3D skeleton action recognition.
2. I suggest creating a new environment for main branch.
3. The bug looks like some dependency library is missing, and the raise error is not accurate. Please upgrade to the latest mmengine, which would provide accurate error info.

**Reply by YuktiADY** [reply]
> 1. Main branch supports both 2D and 3D skeleton action recognition.
> 
>     2. I suggest creating a new environment for main branch.
> 
>     3. The bug looks like some dependency library is missing, and the raise error is not accurate. Please upgrade to the latest mmengine, which would provide accurate error info.

@cir7 Yes i created the new conda environment for main branch. Finally it worked after so many errors when i am training on my custom dataset. Thank you :)

I have few questions:

1. When I read your ntu60_3d.pkl for 3d . There are 6 keys in the .pkl file **'xsub_train', 'xsub_val', 'xset_train', 'xset_val', 'xview_train', 'xview_val'**. In the https://github.com/open-mmlab/mmaction2/blob/main/configs/skeleton/stgcn/stgcn_8xb16-joint-u100-80e_ntu60-xsub-keypoint-3d.py. Only **xsub_train** and **xsub_val**  are used in the config ? Why 'xset_train', 'xset_val', 'xview_train', 'xview_val' are required and where it is referred ? For testing the trained model xsub_val is used  since xsub_val is used for val and test pipeline ?

train_dataloader = dict(
    batch_size=16,
    num_workers=2,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=True),
    dataset=dict(
        type='RepeatDataset',
        times=5,
        dataset=dict(
            type=dataset_type,
            ann_file=ann_file,
            pipeline=train_pipeline,
            **split='xsub_train'**)))
val_dataloader = dict(
    batch_size=16,
    num_workers=2,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type=dataset_type,
        ann_file=ann_file,
        pipeline=val_pipeline,
        **split='xsub_val',**
        test_mode=True))
test_dataloader = dict(
    batch_size=1,
    num_workers=2,
    persistent_workers=True,
    sampler=dict(type='DefaultSampler', shuffle=False),
    dataset=dict(
        type=dataset_type,
        ann_file=ann_file,
        pipeline=test_pipeline,
        **split='xsub_val',**
        test_mode=True))

2. I did the 2D action recognition from 0.x branch . When i see the config for the main branch for the 2d action recognition.
The model definition 'type' is different and train_pipeline is also different. Does the model definition is different in both the branches ??

Can i use the 0.x branch config for 2d action recognition instead of main branch . Will there be any difference in the results ?

**config for main branch** 

 model = dict(
    type='RecognizerGCN',
    backbone=dict(
        type='STGCN', graph_cfg=dict(layout='coco', mode='stgcn_spatial')),
    cls_head=dict(type='GCNHead', num_classes=60, in_channels=256))

train_pipeline = [
    dict(type='PreNormalize2D'),
    dict(type='GenSkeFeat', dataset='coco', feats=['j']),
    dict(type='UniformSampleFrames', clip_len=100),
    dict(type='PoseDecode'),
    dict(type='FormatGCNInput', num_person=2),
    dict(type='PackActionInputs')


**config for 0.x branch** 

model = dict(
    type='SkeletonGCN',
    backbone=dict(
        type='STGCN',
        in_channels=3,
        edge_importance_weighting=True,
        graph_cfg=dict(layout='coco', strategy='spatial')),
    cls_head=dict(
        type='STGCNHead',
        #num_classes=60,
        num_classes=16,
        in_channels=256,
        loss_cls=dict(type='CrossEntropyLoss')),
    train_cfg=None,
    test_cfg=None)

train_pipeline = [
    dict(type='PaddingWithLoop', clip_len=300),
    dict(type='PoseDecode'),
    dict(type='FormatGCNInput', input_format='NCTVM'),
    dict(type='PoseNormalize'),
    dict(type='Collect', keys=['keypoint', 'label'], meta_keys=[]),
    #dict(type='Collect', keys=['images', 'annotations'],meta_keys=[]),
    #dict(type='ToTensor', keys=['images', 'annotations'])
    dict(type='ToTensor', keys=['keypoint'])
]

Can you please tell me about these questions.
Awaiting your response.

Thanks in advance.

**Comment by YuktiADY** [comment]
@cir7 could you please answer this question so that i can proceed further ?

**Comment by cir7** [comment]
1. Remain keys are for other types of evaluation on NTU60, e.g., 'xview_train'  and 'xview_val' for cross-view.
2. We refactor skeleton-based action recognition on the main branch, I don't think the two branches would keep the same results.

**Reply by YuktiADY** [reply]
> 1. Remain keys are for other types of evaluation on NTU60, e.g., 'xview_train'  and 'xview_val' for cross-view.
> 2. We refactor skeleton-based action recognition on the main branch, I don't think the two branches would keep the same results.

@cir7  could you please answer these questions ??
1. If i dont use those keys. Only use xsub_train and xsub_val. It doesnot make any difference in the evaluation results ?
  For testing also xsub_Val is used as val = test pipeline ?

3. Could you please elaborate the second point. What do you mean by it. ??
 
The model definition seems same in the config for the both branches ??

By looking at the config for main branch

model = dict(
type='RecognizerGCN',
backbone=dict(
type='STGCN', graph_cfg=dict(layout='coco', mode='stgcn_spatial')),
cls_head=dict(type='GCNHead', num_classes=60, in_channels=256))

train_pipeline = [
dict(type='PreNormalize2D'),
dict(type='GenSkeFeat', dataset='coco', feats=['j']),
dict(type='UniformSampleFrames', clip_len=100),
dict(type='PoseDecode'),
dict(type='FormatGCNInput', num_person=2),
dict(type='PackActionInputs')

config for 0.x branch

model = dict(
type='SkeletonGCN',
backbone=dict(
type='STGCN',
in_channels=3,
edge_importance_weighting=True,
graph_cfg=dict(layout='coco', strategy='spatial')),
cls_head=dict(
type='STGCNHead',
#num_classes=60,
num_classes=16,
in_channels=256,
loss_cls=dict(type='CrossEntropyLoss')),
train_cfg=None,
test_cfg=None)

**Comment by cir7** [comment]
1. each type of split has independent results and will not affect the others. For most cases, the testing phase uses same data as val phase
2. That means there are large differences between two branches, e.g., train_pipeline as you can see. I can't guarantee the same result

**Reply by YuktiADY** [reply]
> 1. each type of split has independent results and will not affect the others. For most cases, the testing phase uses same data as val phase
> 2. That means there are large differences between two branches, e.g., train_pipeline as you can see. I can't guarantee the same result

@cir7 Thank you for answering the questions.
Could you provide the link where the config parameters description is given so that i can tune the parameters acc to my experiments ?

**Comment by YuktiADY** [comment]
@cir7 I am using the pre-trained model for 3d action . But since the original code uses 25 joints. My custom dataset uses only 17 so i modified the code for 17 joints for my dataset. Now when i try to use pre-trained  model and train on my dataset , which is obvious.
I ran this script for training : 
python tools/train.py configs/skeleton/stgcn/stgcn_8xb16-joint-u100-80e_ntu60-xsub-keypoint-3d_omnilab.py --work-dir ./work_dirs/stgcn_80e_ntu60_xsub_keypoint_3d_25_04_23 --resume stgcn_8xb16-joint-u100-80e_ntu60-xsub-keypoint-3d_20221129-850308e1.pth --seed 0 --deterministic --cfg-options  gpu_ids=[0,1]

How can use your pre trained model for training on my custom dataset.

oads checkpoint by local backend from path: stgcn_8xb16-joint-u100-80e_ntu60-xsub-keypoint-3d_20221129-850308e1.pth
The model and loaded state dict do not match exactly

size mismatch for backbone.data_bn.weight: copying a param with shape torch.Size([75]) from checkpoint, the shape in current model is torch.Size([51]).
size mismatch for backbone.data_bn.bias: copying a param with shape torch.Size([75]) from checkpoint, the shape in current model is torch.Size([51]).
size mismatch for backbone.data_bn.running_mean: copying a param with shape torch.Size([75]) from checkpoint, the shape in current model is torch.Size([51]).
size mismatch for backbone.data_bn.running_var: copying a param with shape torch.Size([75]) from checkpoint, the shape in current model is torch.Size([51]).
size mismatch for backbone.gcn.0.gcn.PA: copying a param with shape torch.Size([3, 25, 25]) from checkpoint, the shape in current model is torch.Size([3, 17, 17]).
size mismatch for backbone.gcn.0.gcn.A: copying a param with shape torch.Size([3, 25, 25]) from checkpoint, the shape in current model is torch.Size([3, 17, 17]).
size mismatch for backbone.gcn.1.gcn.PA: copying a param with shape torch.Size([3, 25, 25]) from checkpoint, the shape in current model is torch.Size([3, 17, 17]).
size mismatch for backbone.gcn.1.gcn.A: copying a param with shape torch.Size([3, 25, 25]) from checkpoint, the shape in current model is torch.Size([3, 17, 17]).
size mismatch for backbone.gcn.2.gcn.PA: copying a param with shape torch.Size([3, 25, 25]) from checkpoint, the shape in current model is torch.Size([3, 17, 17]).
size mismatch for backbone.gcn.2.gcn.A: copying a param with shape torch.Size([3, 25, 25]) from checkpoint, the shape in current model is torch.Size([3, 17, 17]).
size mismatch for backbone.gcn.3.gcn.PA: copying a param with shape torch.Size([3, 25, 25]) from checkpoint, the shape in current model is torch.Size([3, 17, 17]).
size mismatch for backbone.gcn.3.gcn.A: copying a param with shape torch.Size([3, 25, 25]) from checkpoint, the shape in current model is torch.Size([3, 17, 17]).
size mismatch for backbone.gcn.4.gcn.PA: copying a param with shape torch.Size([3, 25, 25]) from checkpoint, the shape in current model is torch.Size([3, 17, 17]).
size mismatch for backbone.gcn.4.gcn.A: copying a param with shape torch.Size([3, 25, 25]) from checkpoint, the shape in current model is torch.Size([3, 17, 17]).
size mismatch for backbone.gcn.5.gcn.PA: copying a param with shape torch.Size([3, 25, 25]) from checkpoint, the shape in current model is torch.Size([3, 17, 17]).
size mismatch for backbone.gcn.5.gcn.A: copying a param with shape torch.Size([3, 25, 25]) from checkpoint, the shape in current model is torch.Size([3, 17, 17]).
size mismatch for backbone.gcn.6.gcn.PA: copying a param with shape torch.Size([3, 25, 25]) from checkpoint, the shape in current model is torch.Size([3, 17, 17]).
size mismatch for backbone.gcn.6.gcn.A: copying a param with shape torch.Size([3, 25, 25]) from checkpoint, the shape in current model is torch.Size([3, 17, 17]).
size mismatch for backbone.gcn.7.gcn.PA: copying a param with shape torch.Size([3, 25, 25]) from checkpoint, the shape in current model is torch.Size([3, 17, 17]).
size mismatch for backbone.gcn.7.gcn.A: copying a param with shape torch.Size([3, 25, 25]) from checkpoint, the shape in current model is torch.Size([3, 17, 17]).
size mismatch for backbone.gcn.8.gcn.PA: copying a param with shape torch.Size([3, 25, 25]) from checkpoint, the shape in current model is torch.Size([3, 17, 17]).
size mismatch for backbone.gcn.8.gcn.A: copying a param with shape torch.Size([3, 25, 25]) from checkpoint, the shape in current model is torch.Size([3, 17, 17]).
size mismatch for backbone.gcn.9.gcn.PA: copying a param with shape torch.Size([3, 25, 25]) from checkpoint, the shape in current model is torch.Size([3, 17, 17]).
size mismatch for backbone.gcn.9.gcn.A: copying a param with shape torch.Size([3, 25, 25]) from checkpoint, the shape in current model is torch.Size([3, 17, 17]).
size mismatch for cls_head.fc.weight: copying a param with shape torch.Size([60, 256]) from checkpoint, the shape in current model is torch.Size([16, 256]).
size mismatch for cls_head.fc.bias: copying a param with shape torch.Size([60]) from checkpoint, the shape in current model is torch.Size([16]).
04/25 18:25:02 - mmengine - INFO - Load checkpoint from stgcn_8xb16-joint-u100-80e_ntu60-xsub-keypoint-3d_20221129-850308e1.pth
04/25 18:25:02 - mmengine - WARNING - The value of random seed in the checkpoint "150966150" is different from the value in `randomness` config "0"
Traceback (most recent call last):
  File "tools/train.py", line 135, in <module>
    main()
  File "tools/train.py", line 131, in main
    runner.train()
  File "/home/yukti/miniconda3/envs/openmmlab/lib/python3.8/site-packages/mmengine/runner/runner.py", line 1694, in train
    self.load_or_resume()
  File "/home/yukti/miniconda3/envs/openmmlab/lib/python3.8/site-packages/mmengine/runner/runner.py", line 1640, in load_or_resume
    self.resume(resume_from)
  File "/home/yukti/miniconda3/envs/openmmlab/lib/python3.8/site-packages/mmengine/runner/runner.py", line 1999, in resume
    self.message_hub.load_state_dict(checkpoint['message_hub'])
KeyError: 'message_hub'


i CHANGED IN GRAPH.PY FILE
FROM 25 JOINTS -->>> 17 JOINTS
elif layout == 'nturgb+d':
            #self.num_node = 25
            self.num_node = 17
            #neighbor_base = [(1, 2), (2, 21), (3, 21), (4, 3), (5, 21), (6, 5),
                             #(7, 6), (8, 7), (9, 21), (10, 9), (11, 10),
                             #(12, 11), (13, 1), (14, 13), (15, 14), (16, 15),
                             #(17, 1), (18, 17), (19, 18), (20, 19), (22, 8),
                             #(23, 8), (24, 12), (25, 12)]
            self.inward = [(15, 13), (13, 11), (16, 14), (14, 12), (11, 5),
                           (12, 6), (9, 7), (7, 5), (10, 8), (8, 6), (5, 0),
                           (6, 0), (1, 0), (3, 1), (2, 0), (4, 2)] 
            #self.inward = [(i - 1, j - 1) for (i, j) in neighbor_base]
            #self.center = 21 - 1
            self.center = 0
Can you please comment on this so that i can proceed further?

When i train using script
Thats#s why i used resume to load pre-trained model

python tools/train.py configs/skeleton/stgcn/stgcn_8xb16-joint-u100-80e_ntu60-xsub-keypoint-3d_omnilab.py --work-dir ./work_dirs/stgcn_80e_ntu60_xsub_keypoint_3d_25_04_23 --load_from stgcn_8xb16-joint-u100-80e_ntu60-xsub-keypoint-3d_20221129-850308e1.pth --seed 0 --deterministic --cfg-options  gpu_ids=[0,1]
usage: train.py [-h] [--work-dir WORK_DIR] [--resume [RESUME]] [--amp] [--no-validate] [--auto-scale-lr] [--seed SEED] [--diff-rank-seed] [--deterministic]
                [--cfg-options CFG_OPTIONS [CFG_OPTIONS ...]] [--launcher {none,pytorch,slurm,mpi}] [--local_rank LOCAL_RANK]
                config
train.py: error: unrecognized arguments: --load_from stgcn_8xb16-joint-u100-80e_ntu60-xsub-keypoint-3d_20221129-850308e1.pth

**Comment by cir7** [comment]
In my opinion, you can't use a pre-train ckpt based on 25 points, for your 17 points data.
